substrate/frame/salary/src/migration.rs

```diff
@@ -0,0 +1,121 @@
+// This file is part of Substrate.
+
+// Copyright (C) Parity Technologies (UK) Ltd.
+// SPDX-License-Identifier: Apache-2.0
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// 	http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//! Storage migrations for the salary pallet.
+
+use super::*;
+use frame_support::{pallet_prelude::*, storage_alias, traits::UncheckedOnRuntimeUpgrade};
+
+mod v0 {
+	use super::*;
+	use frame_system::pallet_prelude::BlockNumberFor as LocalBlockNumberFor;
+
+	// V0 types.
+	pub type CycleIndexOf<T> = LocalBlockNumberFor<T>;
+	pub type StatusOf<T, I> = StatusType<CycleIndexOf<T>, LocalBlockNumberFor<T>, BalanceOf<T, I>>;
+	pub type ClaimantStatusOf<T, I> = ClaimantStatus<CycleIndexOf<T>, BalanceOf<T, I>, IdOf<T, I>>;
+
+	/// V0 alias for [`crate::Status`].
+	#[storage_alias]
+	pub type Status<T: Config<I>, I: 'static> =
+		StorageValue<Pallet<T, I>, StatusOf<T, I>, OptionQuery>;
+
+	/// V0 alias for [`crate::Claimant`].
+	#[storage_alias]
+	pub type Claimant<T: Config<I>, I: 'static> = StorageMap<
+		Pallet<T, I>,
+		Twox64Concat,
+		<T as frame_system::Config>::AccountId,
+		ClaimantStatusOf<T, I>,
+		OptionQuery,
+	>;
+}
+
+pub mod v1 {
+	use super::{BlockNumberFor as NewBlockNumberFor, *};
+	use frame_system::pallet_prelude::BlockNumberFor as LocalBlockNumberFor;
+
+	/// Converts previous (local) block number into the new one. May just be identity functions
```

Comment:
```suggestion
	/// Converts old block number into the new one. May just be identity functions
```
Is there an inherent local/remote duality, or is it just old/new?

Reply:
Inherent local -> whatever you've now decided (new)

In the context of this migration because the salary pallet previously did not allow selection of a block provider, it defaulted frame_system's, which is always local I believe. However, @gui1117 did point out this code may be copy pasted so if you think generalizing it is best, I can make the change 👍 Just let me know.